util/start_test

```diff
@@ -676,7 +676,11 @@ def set_up_environment():
         args.performance = True
         args.gen_graphs = True
         args.compopts += " --fast"
-        if not tgt_platform == "darwin":
+        # Force static linking unless the target requires dynamic.
```

Comment:
Reversing the condition to check it (and using liberal pseudocode): we won't force static linking iff:
  platform == darwin || (cray-prgenv && (isSet(CRAY_MEMKIND) || locModel!=flat))

That seems okay, but what about locModel!=flat when using something other than a cray-prgenv compiler?  Wouldn't we also want to use libnuma and thus dynamic linking if we were testing locModel=numa with hwloc on non-Cray linux64 systems?  To achieve this, should we change make/compiler/Makefile.<several-other-compilers> similarly to how we changed make/compiler/Makefile.cray-prgenv?  (But maybe this is a longer-term question and we should tackle it as numa matures?)